src/App.tsx

```diff
@@ -1,7 +1,33 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string;
+};
+
+class App extends React.Component <{}, State> {
+  state = { pressedKey: 'Nothing was pressed yet' };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleKeyboardEvent);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyboardEvent);
+  }
+
+  handleKeyboardEvent = (event: KeyboardEvent) => {
+    this.setState({
+      pressedKey: `The last pressed key is [${event.key}]`,
```

Comment:
`The last pressed key is` doesn’t change, so you can save just event.key on state